esmerald/contrib/schedulers/asyncz/decorator.py

```diff
@@ -55,9 +57,23 @@ async def my_task():
             print("Task executed every 10 seconds")
     """
 
-    def wrapper(func: F) -> Task:
+    def decorator(func: Callable) -> "Task":
+        if is_async_callable(func):
+
+            @wraps(func)
+            async def wrapper(*args: Any, **kwargs: Any) -> Any:
+                return await func(*args, **kwargs)
+
+        else:
+
+            @wraps(func)
+            def wrapper(*args: Any, **kwargs: Any) -> Any:
+                return func(*args, **kwargs)
+
+        wrapper.__original_func__ = func  # Attach original function
+
```

Comment:
why? We have `__wrapped__` by wraps. And anyway, wouldn't it be better to mark the wrapper async in case of an async function (possible, see asgiref)?